working/0546-patterns/patterns-feature-specification.md

```diff
@@ -1314,12 +1402,15 @@ The context type schema for a pattern `p` is:
     var [[int x]!] = [[]]; // Infers List<List<int>?> for the list literal.
     ```
 
-*   **Literal** or **constant**: The context type schema is the static type of
-    the pattern's constant value expression.
+*   **Constant**: The context type schema is the static type of the pattern's
+    value.
 
 *   **Variable**:
 
-    1.  If `p` has no type annotation, the context type schema is `?`.
+    1.  In an assignment context, the context type schema is the static type of
+        the variable that `p` resolves to.
+
```

Comment:
Yes. A `variablePattern` always refers to a variable and never a constant. A `constantPattern` always refers to a constant. Those two kinds of patterns are syntactically ambiguous since a short identifier is valid for both, but we resolve that syntactically using the surrounding pattern context which is known at parse time (which is why I moved the "pattern context" section up before the static semantics).